be/src/agent/heartbeat_server.cpp

```diff
@@ -119,6 +121,13 @@ Status HeartbeatServer::_heartbeat(
                          << " local epoch: " << _epoch << " received epoch: " << master_info.epoch;
             return Status("epoch is not greater than local. ignore heartbeat.");
         }
+    } else {
+        // when Master FE restarted, host and port remains the same, but epoch will be increased.
+        if (master_info.epoch > _epoch) {
+            _epoch = master_info.epoch;
```

Comment:
this function may be called concurrently. and changing `_epoch` and testing `_epoch` without any protect would lead to strange result

Reply:
Add a mutex to protect _heartbeat() function